openssl/src/crypto/rsa.rs

```diff
@@ -61,6 +74,19 @@ impl RSA {
         }
     }
 
+    /// set the key modulus
+    pub fn set_n(&mut self, n: BigNum) {
+        unsafe {
+            (*self.0).n = n.into_raw();
```

Comment:
By asking this question, you're making me realize that I don't actually want to be able to set these fields after construction. I should change the new() constructor to from_public_components(e,n) {..}

In that case the pkey wouldn't have a reference to the RSA key yet, which I think would remove the need for the lock on these set functions. It would remove the potential from the RSA struct changing out from under anyone, right?